config/filter.d/screensharingd.conf

```diff
@@ -0,0 +1,33 @@
+# Fail2Ban configuration file
+#
+# Author: Simon Brown
+#
+# $Revision: 1 $ 
+#
+# Filter for Mac OS X Screen Sharing service
+
+[INCLUDES]
+
+# Read common prefixes. If any customizations available -- read them from
+# common.local
+before = common.conf
+
+
+[Definition]
+
+_daemon = screensharingd
+
+# Option:  failregex
+# Notes.:  regex to match the password failures messages in the logfile. The
+#          host must be matched by a group named "host". The tag "<HOST>" can
+#          be used for standard IP/hostname matching and is only an alias for
+#          (?:::f{4,6}:)?(?P<host>[\w\-.^_]+)
+# Values:  TEXT
+#
+failregex = ^.+ screensharingd.+: Authentication: FAILED :: User Name: .+ :: Viewer Address: <HOST> :: .*$
```

Comment:
Please do not use not hard-anchored at end (`.*$`) regexp together with greedy catch-all (`.*`). 
Especially dangerous is it if the greedy catch-all take place before <HOST> - it is a potential weak spot resp. even vulnerability.
Simple way to fix it is to change it to non-greedy catch-all (`.+?`) but it makes search of match slower.
Better would be to state more precisely your regular expression as exact as possible resp. to resolve a not hard-anchored end (`.*` before `$`) in this regexp.